File: src/test/java/com/kobylynskyi/graphql/codegen/scala/compile-pass/VersionQueryResponse.scala
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResult
import java.util.{ Map => JMap }

/**
 * Version of the application.
 */
class VersionQueryResponse extends GraphQLResult[JMap[String, String]] {

    /**
     * Version of the application.
     */
    def version(): String = {
        val data: JMap[String, String] = getData
        if (data != null) data.get(VersionQueryResponse.OPERATION_NAME) else null
    }

}

object VersionQueryResponse {

    private final val OPERATION_NAME: String = "version"

}

File: src/test/java/com/kobylynskyi/graphql/codegen/kotlin/compile-pass/VersionQueryResponse.kt
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResult

/**
 * Version of the application.
 */
open class VersionQueryResponse : GraphQLResult<MutableMap<String, String>>() {

    companion object {
        const val OPERATION_NAME: String = "version"
    }

    /**
     * Version of the application.
     */
    fun version(): String {
        val data: MutableMap<String, String> = super.getData()
        return data.getValue(OPERATION_NAME)
    }
}
